master/internal/db/postgres_experiments.go

```diff
@@ -35,3 +38,148 @@ func (db *PgDB) ExperimentLabelUsage() (labelUsage map[string]int, err error) {
 	}
 	return labelUsage, nil
 }
+
+// MetricNames returns the set of training and validation metric names that have been recorded for
+// an experiment.
+func (db *PgDB) MetricNames(experimentID int, sStartTime time.Time, vStartTime time.Time) (
```

Comment:
I have mixed feelings on that one, actually - they're never queried separately. I've gone ahead and split up the trainingBatches / validationBatches functions, but since there's so much commonality between the queries I don't want them to end up too far apart visually, in case one is edited for whatever reason. I prefer these together, but if you feel strongly let's chat.

Reply:
For some context on the way I view it, this actually was originally one query with a column for validation metric names and one for training metric names, but there was still some duplication since each combination of names in each row was considered distinct. So I split it up to avoid having MORE deduplication code in Go. But the queries are VERY closely related and again, always used together.

Reply:
Yeah, I understand, if they should never be used separately this is definitely fine. I was more thinking the signature had gotten a little complex was all.